### PR TITLE
Adding RecordDatastoreSegment to .NET Agent API

### PR DESCRIPTION
**Pull request by chynesNR**
PR where the feature was added: https://github.com/newrelic/newrelic-dotnet-agent/pull/2320

### CODE REVIEW

**Comment by chynesNR** [comment]
The page that was updated: https://deploy-preview-18999--docs-website-netlify.netlify.app/docs/apm/agents/net-agent/net-agent-api/net-agent-api/

**Comment by jeff-colucci** [comment]
@chynesNR I see there's some "string" entries that have question marks after them, is this intended or are we waiting for confirmation that string is correct before we merge it? Just let me know how you'd like it handled and I'll get it taken care of!

**Comment by chynesNR** [comment]
> @chynesNR I see there's some "string" entries that have question marks after them, is this intended or are we waiting for confirmation that string is correct before we merge it? Just let me know how you'd like it handled and I'll get it taken care of!

It's intentional. That does look a little funny if you aren't familiar with C# syntax. It means a "nullable string", i.e. it can be either a string value or null.

**Review by jeff-colucci — APPROVED** [review]

**Pull request merged**
